src/metabase/models/collection.clj

```diff
@@ -67,6 +67,18 @@
   (derive ::mi/read-policy.full-perms-for-perms-set)
   (derive ::mi/write-policy.full-perms-for-perms-set))
 
+(defmethod mi/can-write? Collection
+  ([instance]
+   (if (= (:entity_id instance) (perms/default-audit-collection-entity-id))
+     false
+     (mi/current-user-has-full-permissions? :write instance)))
+  ([model pk]
+   (if (and
+        (string? (perms/default-audit-collection-entity-id))
+        (= pk (t2/select-one-fn :id 'Collection :entity_id (perms/default-audit-collection-entity-id))))
```

Comment:
Since you're hitting the DB anyway, how about you just fetch the collection here and then call `can-write?` again with the instance as the arg? Then you don't have to have the logic duplicated for the different arities